### PR TITLE
bpo-40489: Add test case for dict contain use after free

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue40489

### CODE REVIEW

File: Lib/test/test_dict.py [hunk 1]
```diff
@@ -1333,6 +1346,7 @@ def __hash__(self):
         pair = [X(), 123]
         dict([pair])
 
+
```
Comment (remilapeyre) [inline Lib/test/test_dict.py]:
```suggestion
```

Comment (corona10) [reply Lib/test/test_dict.py]: Thanks :)

**Review by remilapeyre — APPROVED** [review]
I tried removing the `Py_INCREF` in `lookdict()` and this test work to expose the segfault :)

**Review by rhettinger — APPROVED** [review]

**Comment by corona10** [comment]
@rhettinger Thank you for the review :)

**Pull request merged**